testing/tcphandler/main.go

```diff
@@ -37,5 +43,23 @@ func handleRequest(conn net.Conn) {
 		log.Print("Error reading: ", err.Error())
 		return
 	}
+	if *file != "" {
+		event := &types.Event{}
+		err = json.Unmarshal(bytes.Trim(buf, "\x00"), &event)
+		if err != nil {
+			log.Print("Recieved non-event: ", err.Error())
+			return
+		}
+		eventJSON, err := json.Marshal(event)
+		if err != nil {
+			log.Print("Error marshalling event: ", err.Error())
+			return
+		}
+		err = ioutil.WriteFile(*file, eventJSON, 0644)
```

Comment:
The problem with `ioutil.WriteFile()` is that it truncates the given file before writing. Would appending to the file be better?

My concern is that, if you receive 2 or more events in a short time period only the last one will be in the file, potentially creating confusion where one might think the "wrong" event was received while it was in fact "swallowed".

Reply:
I see, but doesn't this introduce a form of race condition?
Here is what I imagine can happen:
- the qa test makes it so that the tcp handler receives an expected event, say `event1`
- while the qa test goes to check what was written to the file, another event, say `event2` arrives and overwrite the content of the output file
- the qa test is considered a failure because `event1 != event2`

Or can this situation never occur because of some other mechanism I'm not aware of? Sorry, I don't have much context on where/how this is going to be used!

Reply:
That's a valid point. For the test I'm concerned with, and the only test invoking this specific handler, the event should be almost identical. The test is only checking for certain key values such as event.check.status and event.check.output.

Let's discuss it when I push the tcp handler to sensu-staging.. it will have much more context regarding the tests.